File: src/main/java/client/inventory/slots/ItemSlot.java
package client.inventory.slots;

import java.util.Arrays;

public abstract class ItemSlot implements Cloneable {

    protected int templateId;
    protected long cashItemSN;
    protected long expire;
    protected byte[] uuid;
    private boolean newItem = true;

    public int getTemplateId() {
        return templateId;
    }

    public void setTemplateId(int templateId) {
        this.templateId = templateId;
    }

    public long getCashItemSN() {
        return cashItemSN;
    }

    public void setCashItemSN(long cashItemSN) {
        this.cashItemSN = cashItemSN;
    }

    public long getExpire() {
        return expire;
    }

    public void setExpire(long expire) {
        this.expire = expire;
    }

    public byte[] getUuid() {
        return uuid;
    }

    public void setUuid(byte[] uuid) {
        this.uuid = uuid;
    }

    public boolean isNewItem() {
        return newItem;
    }

    public void setNewItem(boolean newItem) {
        this.newItem = newItem;
    }

    @Override
    public String toString() {
        return "ItemSlot{" +
                "templateId=" + templateId +
                ", cashItemSN=" + cashItemSN +
                ", expire=" + expire +
                ", uuid=" + Arrays.toString(uuid) +
                '}';
    }

    @Override
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

File: src/main/java/client/inventory/item/slots/ItemSlot.kt
package client.inventory.item.slots

import kotlin.properties.Delegates
import kotlin.properties.ReadWriteProperty

abstract class ItemSlot {

    protected fun <T> getObservableValue(default: T): ReadWriteProperty<Any?, T> {
        return Delegates.observable(default) { _, _, _ -> updated = true }
    }

    var templateId = 0
    var cashItemSN: Long = 0
    var expire: Long = 0
    var uuid: ByteArray? = null
    var isNewItem = true
    var updated = false

    override fun toString(): String {
        return "ItemSlot{templateId=$templateId, cashItemSN=$cashItemSN, expire=$expire, uuid=${uuid?.contentToString()}}"
    }
}